## Aim

After having classified papers, I need to move on to split task for affiliation coding. Before that, I checked the validity of the paper type classification result. 

It shows that five articles do not have author names. I corrected six paper classificaiton results. These are the results before i corrected them. 

I also found 100 papers (138 authors) that do not have affiliation strings. I also need to remove them from the pool of "research papers". 

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('../../data/processed/research_author_with_pred.csv')

In [3]:
df.shape

(11476, 39)

In [4]:
df[df.authorFullName.isnull()]
# This doi is M as well: 10.1111/j.1460-2466.2000.tb02849.x, and I have corrected these six

authorID  \
3170  10.1111/j.1460-2466.1987.tb00988.x+nan   
3451  10.1111/j.1460-2466.1982.tb02518.x+nan   
4978  10.1111/j.1460-2466.1958.tb01157.x+nan   
4982  10.1111/j.1460-2466.1958.tb01147.x+nan   
7774  10.1111/j.1468-2958.1975.tb00266.x+nan   

                                     doi  \
3170  10.1111/j.1460-2466.1987.tb00988.x   
3451  10.1111/j.1460-2466.1982.tb02518.x   
4978  10.1111/j.1460-2466.1958.tb01157.x   
4982  10.1111/j.1460-2466.1958.tb01147.x   
7774  10.1111/j.1468-2958.1975.tb00266.x   

                                                    url  year  \
3170  https://academic.oup.com/joc/article/37/2/191/...  1987   
3451  https://academic.oup.com/joc/article/32/4/55/4...  1982   
4978  https://academic.oup.com/joc/article/8/4/201/4...  1958   
4982  https://academic.oup.com/joc/article/8/3/143/4...  1958   
7774  https://academic.oup.com/hcr/article/1/2/182/4...  1975   

                                                  title  \
3170                                      Briefly noted   
3451  The “New Technology”: Who Sells It? Who Needs ...   
4978                                            Reviews   
4982                                            Reviews   
7774  Handbook of Communication: The Parts and the W...   

                           journal  numberOfAuthors  authorPosition  \
3170      Journal of Communication              NaN             NaN   
3451      Journal of Communication              NaN             NaN   
4978      Journal of Communication              NaN             NaN   
4982      Journal of Communication              NaN             NaN   
7774  Human Communication Research              NaN             NaN   

     authorFullName firstName  ...     white raceHighest raceSecondHighest  \
3170            NaN       NaN  ...  0.384294    0.461375          0.384294   
3451            NaN       NaN  ...  0.384294    0.461375          0.384294   
4978            NaN       NaN  ...  0.384294    0.461375          0.384294   
4982            NaN       NaN  ...  0.384294    0.461375          0.384294   
7774            NaN       NaN  ...  0.384294    0.461375          0.384294   

      raceDiff affProcessed  affiliation.2  ROR_AFFNAME matchMethod ROR_ID  \
3170  0.077081          NaN            NaN          NaN         NaN    NaN   
3451  0.077081          NaN            NaN          NaN         NaN    NaN   
4978  0.077081          NaN            NaN          NaN         NaN    NaN   
4982  0.077081          NaN            NaN          NaN         NaN    NaN   
7774  0.077081          NaN            NaN          NaN         NaN    NaN   

     type  
3170    R  
3451    R  
4978    R  
4982    R  
7774    R  

[5 rows x 39 columns]

In [7]:
df[df.authorFullName.isnull()].doi

3170    10.1111/j.1460-2466.1987.tb00988.x
3451    10.1111/j.1460-2466.1982.tb02518.x
4978    10.1111/j.1460-2466.1958.tb01157.x
4982    10.1111/j.1460-2466.1958.tb01147.x
7774    10.1111/j.1468-2958.1975.tb00266.x
Name: doi, dtype: object

These are the five articles that do not have author names. They are clearly misclassified by us:

3170    10.1111/j.1460-2466.1987.tb00988.x \ \
3451    10.1111/j.1460-2466.1982.tb02518.x \ \
4978    10.1111/j.1460-2466.1958.tb01157.x \ \
4982    10.1111/j.1460-2466.1958.tb01147.x \ \
7774    10.1111/j.1468-2958.1975.tb00266.x

In [8]:
# These are the papers that do not have affiliations. 
df[df['affiliation.2'].isnull()]

authorID  \
142                  10.1093/joc/jqab027+14.0   
917                    10.1111/jcom.12153+3.0   
2476   10.1111/j.1460-2466.2000.tb02849.x+1.0   
2477   10.1111/j.1460-2466.2000.tb02849.x+2.0   
3149   10.1111/j.1460-2466.1987.tb00995.x+1.0   
...                                       ...   
10148    10.1111/j.1083-6101.2006.00318.x+3.0   
10149    10.1111/j.1083-6101.2006.00318.x+4.0   
10175    10.1111/j.1083-6101.2006.00305.x+1.0   
10217    10.1111/j.1083-6101.2006.00021.x+1.0   
10793  10.1111/j.1083-6101.1996.tb00185.x+1.0   

                                      doi  \
142                   10.1093/joc/jqab027   
917                    10.1111/jcom.12153   
2476   10.1111/j.1460-2466.2000.tb02849.x   
2477   10.1111/j.1460-2466.2000.tb02849.x   
3149   10.1111/j.1460-2466.1987.tb00995.x   
...                                   ...   
10148    10.1111/j.1083-6101.2006.00318.x   
10149    10.1111/j.1083-6101.2006.00318.x   
10175    10.1111/j.1083-6101.2006.00305.x   
10217    10.1111/j.1083-6101.2006.00021.x   
10793  10.1111/j.1083-6101.1996.tb00185.x   

                                                     url  year  \
142    https://academic.oup.com/joc/article/71/5/803/...  2021   
917    https://academic.oup.com/joc/article/65/3/443/...  2015   
2476   https://academic.oup.com/joc/article/50/3/4/41...  2000   
2477   https://academic.oup.com/joc/article/50/3/4/41...  2000   
3149   https://academic.oup.com/joc/article/37/3/68/4...  1987   
...                                                  ...   ...   
10148  https://academic.oup.com/jcmc/article/12/1/136...  2006   
10149  https://academic.oup.com/jcmc/article/12/1/136...  2006   
10175  https://academic.oup.com/jcmc/article/11/4/101...  2006   
10217  https://academic.oup.com/jcmc/article/11/2/442...  2006   
10793  https://academic.oup.com/jcmc/article/2/1/JCMC...  1996   

                                                   title  \
142    Decolonizing Open Science: Southern Interventions   
917    Parent and Child Media Exposure, Preschooler D...   
2476   Elderly Lifestyles in the 21st Century: "Doris...   
2477   Elderly Lifestyles in the 21st Century: "Doris...   
3149           Calculating Risk: Radiation and Chernobyl   
...                                                  ...   
10148  Managing Impressions in a Virtual Environment:...   
10149  Managing Impressions in a Virtual Environment:...   
10175  Gender and the Use of Exclamation Points in Co...   
10217  Ouch!: an Examination of the Self-Representati...   
10793  Virtual(ly) Law: the Emergence of Law in Lambd...   

                                          journal  numberOfAuthors  \
142                      Journal of Communication             23.0   
917                      Journal of Communication              3.0   
2476                     Journal of Communication              2.0   
2477                     Journal of Communication              2.0   
3149                     Journal of Communication              1.0   
...                                           ...              ...   
10148  Journal of Computer-Mediated Communication              4.0   
10149  Journal of Computer-Mediated Communication              4.0   
10175  Journal of Computer-Mediated Communication              1.0   
10217  Journal of Computer-Mediated Communication              1.0   
10793  Journal of Computer-Mediated Communication              1.0   

       authorPosition       authorFullName firstName  ...     white  \
142              14.0  Gayle Moana Johnson     Gayle  ...  0.608817   
917               3.0                            NaN  ...  0.384294   
2476              1.0       Angie Williams     Angie  ...  0.465989   
2477              2.0  Virpi Ylänne-McEwen     Virpi  ...  0.924163   
3149              1.0    Robert Peter Gale    Robert  ...  0.882981   
...               ...                  ...       ...  ...       ...   
10148             3.0      Tiffany Charles   Tiffany  .